### PR TITLE
Created example config for Apache Guacamole app

### PR DESCRIPTION
**Pull request by MickMorley**

### CODE REVIEW

**Review by nitnelave — CHANGES_REQUESTED** [review]

File: example_configs/apacheguacamole.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+!! IMPORTANT - LDAP only works with LLDAP if using a [database authentication](https://guacamole.apache.org/doc/gug/ldap-auth.html#associating-ldap-with-a-database).  The Apache Guacamole does support using LDAP to store user config but that is not in scope here.
```
[the remark this answers is not in the document]
Comment (MickMorley) [reply example_configs/apacheguacamole.md:1]: Can do

File: example_configs/apacheguacamole.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+*Apache Guacamole does not lock you out when enabling LDAP.  Your `static` IDs still are able to log in.
+
+##  To enable LDAP
+Restart your Apache Guacamole app for changes to take effect
+
+## To enable users
+Before logging in, you have to manually create users in the Apache Guacamole application in order for permissions/connections/etc for the users to be set.
+
+Using your static ID, create a username that matches your LDAP user. If applicable, tick the permissions and/or connections that you want this user to see.
```
[the remark this answers is not in the document]
Comment (MickMorley) [reply example_configs/apacheguacamole.md]: Thanks, I've added more clarity.

File: example_configs/apacheguacamole.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+ldap-hostname: localhost
+ldap-port: 3890
+ldap-user-base-dn: ou=people,dc=example,dc=com
+ldap-username-attribute: uid
+ldap-search-bind-dn: uid=admin,ou=people,dc=example,dc=com
+ldap-search-bind-password: replacewithyoursecret
+ldap-user-search-filter: (memberof=cn=lldap_apacheguac,ou=groups,dc=example,dc=com)
+```
+
+*Exclude `ldap-user-search-filter` if you do not want to limit users based on a group(s)
```
Comment (nitnelave) [inline example_configs/apacheguacamole.md:25]: Nit: this markdown doesn't render properly, you need a space after the *
You can click on the three dots in the top-right corner of the file and "View the file" to see the markdown rendered.

File: example_configs/apacheguacamole.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+*Replace `dc=example,dc=com` with your LLDAP configured domain for all occurances
+
+*Apache Guacamole does not lock you out when enabling LDAP.  Your `static` IDs still are able to log in.
+
+##  To enable LDAP
+Restart your Apache Guacamole app for changes to take effect
+
+## To enable users
+Before logging in with an LLDAP user, you have to manually create it using your static ID in Apache Guacamole. This applies to each user that you want to log in with using LDAP authentication.  Otherwise the user will be logged in without any permissions/connections/etc.
```
Comment (nitnelave) [inline example_configs/apacheguacamole.md:35]: I assume there's no automated way to import all users?

File: example_configs/apacheguacamole.md (hunk 3 above)
Comment (MickMorley) [reply example_configs/apacheguacamole.md:25]: Thank you, I was actually going for one of those * wildcard caveats that you see when reading a passage or nutrition facts. 

I changed it to bullets as you suggested.

Comment (nitnelave) [reply example_configs/apacheguacamole.md:25]: I don't see the change, are you sure you made it? Did you commit?

**Comment by martadinata666** [comment]
In mean time, can you also add LDAP config on docker guacamole

Technically just add ENV to guacamole 
```
       - LDAP_HOSTNAME=192.168.0.2 #IP or hostname
       - LDAP_PORT=3890 #LLDAP port
       - LDAP_ENCRYPTION_METHOD=none #disabling SSL
       - LDAP_SEARCH_BIND_DN=uid=admin,ou=people,dc=example,dc=com #
       - LDAP_SEARCH_BIND_PASSWORD=thesecret #LLDAP_LDAP_USER_PASS env
       - LDAP_USER_BASE_DN=ou=people,dc=example,dc=com
       - LDAP_USER_SEARCH_FILTER=(memberof=cn=lldap_guacamole,ou=groups,dc=example,dc=com) #filter with `lldap_guacamole` sample group.
```

**Review by nitnelave — APPROVED** [review]

**Pull request merged**